### PR TITLE
CHE-1935; add custom User-Agent header value for JGit;

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?

 Sets custom header instead of "JGit-4.3.0" for git providers have filtration by User-Agent header.
### What issues does this PR fix or reference?
#1935  codenvy/codenvy#430
### Tests written?

No
### Docs requirements?

None

### CODE REVIEW

File: wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java [hunk 1]
```diff
@@ -1605,7 +1608,7 @@ public void configure(Transport transport) {
             }
 
             ProxyAuthenticator.initAuthenticator(remoteUrl);
-
+            UserAgent.set(USER_AGENT);
```
Comment (benoitf) [inline wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java]: do we need to call this static method from JGitConnection instance or from a class that is less created like JGitConnectionFactory ?

Comment (mshaposhnik) [inline wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java]: Yes, make sense. Maybe i should move it to connection constructor.

**Comment by benoitf** [comment]
Not in factory ? So we share for all connections

**Comment by skabashnyuk** [comment]
> Not in factory ? Son we share for all connections

@benoitf  Yes. What's wrong with this? How do you propose?

**Comment by benoitf** [comment]
@skabashnyuk with latest change it's ok for me (it's in the factory so we call only once)

**Pull request merged**